diff --git a/modules/juce_events/native/juce_linux_Messaging.cpp b/modules/juce_events/native/juce_linux_Messaging.cpp
index 48e28a7..b17aafc 100644
--- a/modules/juce_events/native/juce_linux_Messaging.cpp
+++ b/modules/juce_events/native/juce_linux_Messaging.cpp
@@ -358,8 +358,13 @@ bool MessageManager::postMessageToSystemQueue (MessageManager::MessageBase* cons
     if (LinuxErrorHandling::errorOccurred)
         return false;
 
-    InternalMessageQueue::getInstanceWithoutCreating()->postMessage (message);
-    return true;
+    if (InternalMessageQueue* const queue = InternalMessageQueue::getInstanceWithoutCreating())
+    {
+        InternalMessageQueue::getInstanceWithoutCreating()->postMessage (message);
+        return true;
+    }
+
+    return false;
 }
 
 void MessageManager::broadcastMessage (const String& /* value */)
