diff --git a/modules/juce_events/native/juce_linux_Messaging.cpp b/modules/juce_events/native/juce_linux_Messaging.cpp
index b2eaece..6e17abb 100644
--- a/modules/juce_events/native/juce_linux_Messaging.cpp
+++ b/modules/juce_events/native/juce_linux_Messaging.cpp
@@ -360,8 +360,13 @@ bool MessageManager::postMessageToSystemQueue (MessageManager::MessageBase* cons
     if (LinuxErrorHandling::errorOccurred)
         return false;
 
-    InternalMessageQueue::getInstanceWithoutCreating()->postMessage (message);
-    return true;
+    if (InternalMessageQueue* const queue = InternalMessageQueue::getInstanceWithoutCreating())
+    {
+        queue->postMessage (message);
+        return true;
+    }
+
+    return false;
 }
 
 void MessageManager::broadcastMessage (const String& /* value */)
