diff --git a/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp b/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp
index 29df141..bc4b02a 100644
--- a/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp
+++ b/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp
@@ -282,10 +282,6 @@ public:
         setInitialDelay (filter->getLatencySamples());
         programsAreChunks (true);
 
-        // NB: For reasons best known to themselves, some hosts fail to load/save plugin
-        // state correctly if the plugin doesn't report that it has at least 1 program.
-        jassert (af->getNumPrograms() > 0);
-
         activePlugins.add (this);
     }
 
@@ -1166,8 +1162,10 @@ public:
                 editorComp->addToDesktop (0, ptr);
                 hostWindow = (HWND) ptr;
               #elif JUCE_LINUX
-                editorComp->addToDesktop (0, ptr);
+                editorComp->addToDesktop (0);
                 hostWindow = (Window) ptr;
+                Window editorWnd = (Window) editorComp->getWindowHandle();
+                XReparentWindow (display, editorWnd, hostWindow, 0, 0);
               #else
                 hostWindow = attachComponentToWindowRef (editorComp, ptr, useNSView);
               #endif
diff --git a/modules/juce_audio_plugin_client/utility/juce_PluginUtilities.cpp b/modules/juce_audio_plugin_client/utility/juce_PluginUtilities.cpp
index 82e7e47..8400493 100644
--- a/modules/juce_audio_plugin_client/utility/juce_PluginUtilities.cpp
+++ b/modules/juce_audio_plugin_client/utility/juce_PluginUtilities.cpp
@@ -31,7 +31,6 @@
 #include "AppConfig.h"
 
 #include "../utility/juce_CheckSettingMacros.h"
-#include "juce_IncludeModuleHeaders.h"
 
 #if _MSC_VER || JUCE_MINGW
 
