diff --git a/modules/juce_core/native/juce_BasicNativeHeaders.h b/modules/juce_core/native/juce_BasicNativeHeaders.h
index cdaa0ef..55c0c8c 100644
--- a/modules/juce_core/native/juce_BasicNativeHeaders.h
+++ b/modules/juce_core/native/juce_BasicNativeHeaders.h
@@ -87,7 +87,7 @@
  #define STRICT 1
  #define WIN32_LEAN_AND_MEAN 1
  #if JUCE_MINGW
-  #define _WIN32_WINNT 0x0501
+  #define _WIN32_WINNT 0x0502
  #else
   #define _WIN32_WINNT 0x0600
  #endif
diff --git a/modules/juce_events/messages/juce_Initialisation.h b/modules/juce_events/messages/juce_Initialisation.h
index 4a93b0f..1b33fd7 100644
--- a/modules/juce_events/messages/juce_Initialisation.h
+++ b/modules/juce_events/messages/juce_Initialisation.h
@@ -89,7 +89,7 @@ public:
 
 #else
  #if JUCE_WINDOWS
-  #if defined (WINAPI) || defined (_WINDOWS_)
+  #if defined (WINAPI) || defined (_WINDOWS_) || defined(JUCE_MINGW)
    #define JUCE_MAIN_FUNCTION       int __stdcall WinMain (HINSTANCE, HINSTANCE, const LPSTR, int)
   #elif defined (_UNICODE)
    #define JUCE_MAIN_FUNCTION       int __stdcall WinMain (void*, void*, const wchar_t*, int)
