diff --git a/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp b/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
index 1bef497..cb4d884 100644
--- a/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
+++ b/modules/juce_audio_processors/format_types/juce_VST3PluginFormat.cpp
@@ -1355,7 +1355,8 @@ private:
 
             const Result result (finder.findDescriptionsAndPerform (f));
 
-            if (result.getErrorMessage() == MatchingDescriptionFinder::getSuccessString())
+            if (result.getErrorMessage() == MatchingDescriptionFinder::getSuccessString() ||
+                result.getErrorMessage().isEmpty())
             {
                 name = description.name;
                 return true;
