diff --git a/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp b/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp
index 4c3df9e46..70ca644e7 100644
--- a/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp
+++ b/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp
@@ -166,6 +166,7 @@ private:
         void operator() (LPWSTR ptr) const noexcept { CoTaskMemFree (ptr); }
     };
 
+   #if JUCE_MSVC
     bool showDialog (IFileDialog& dialog, bool async)
     {
         FILEOPENDIALOGOPTIONS flags = {};
@@ -383,6 +384,7 @@ private:
 
         return result;
     }
+   #endif
 
     Array<URL> openDialogPreVista (bool async)
     {
@@ -499,11 +501,13 @@ private:
 
         const Remover remover (*this);
 
+       #if JUCE_MSVC
         if (SystemStats::getOperatingSystemType() >= SystemStats::WinVista
             && customComponent == nullptr)
         {
             return openDialogVistaAndUp (async);
         }
+       #endif
 
         return openDialogPreVista (async);
     }
