diff --git a/modules/juce_opengl/native/juce_OpenGLExtensions.h b/modules/juce_opengl/native/juce_OpenGLExtensions.h
index 142a4688d..674321c63 100644
--- a/modules/juce_opengl/native/juce_OpenGLExtensions.h
+++ b/modules/juce_opengl/native/juce_OpenGLExtensions.h
@@ -83,7 +83,14 @@ namespace juce
     X (glCheckFramebufferStatus) \
     X (glFramebufferTexture2D) \
     X (glFramebufferRenderbuffer) \
-    X (glGetFramebufferAttachmentParameteriv)
+    X (glGetFramebufferAttachmentParameteriv) \
+    X (glTransformFeedbackVaryings) \
+    X (glBeginTransformFeedback) \
+    X (glEndTransformFeedback) \
+    X (glBindBufferBase) \
+    X (glMapBufferRange) \
+    X (glUnmapBuffer)
+ 
 
 /** @internal This macro contains a list of GL extension functions that need to be dynamically loaded on Windows/Linux.
     @see OpenGLExtensionFunctions
