diff --git a/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp b/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp
index ed6688693..ba64e8b2c 100644
--- a/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp
+++ b/modules/juce_audio_processors/format_types/juce_LADSPAPluginFormat.cpp
@@ -18,7 +18,7 @@
 
 #if JUCE_PLUGINHOST_LADSPA && (JUCE_LINUX || JUCE_BSD)
 
-#include <ladspa.h>
+#include "ladspa.h"
 
 namespace juce
 {
