diff --git a/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp b/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp
index ec946b7a4..ce42c0093 100644
--- a/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp
+++ b/modules/juce_gui_basics/native/juce_win32_FileChooser.cpp
@@ -159,6 +159,7 @@ private:
         void operator() (LPWSTR ptr) const noexcept { CoTaskMemFree (ptr); }
     };
 
+   #if JUCE_MSVC
     bool showDialog (IFileDialog& dialog, bool async)
     {
         FILEOPENDIALOGOPTIONS flags = {};
@@ -376,6 +377,7 @@ private:
 
         return result;
     }
+   #endif
 
     Array<URL> openDialogPreVista (bool async)
     {
@@ -492,11 +494,13 @@ private:
 
         const Remover remover (*this);
 
+       #if JUCE_MSVC
         if (SystemStats::getOperatingSystemType() >= SystemStats::WinVista
             && customComponent == nullptr)
         {
             return openDialogVistaAndUp (async);
         }
+       #endif
 
         return openDialogPreVista (async);
     }
