diff --git a/modules/juce_core/native/juce_posix_SharedCode.h b/modules/juce_core/native/juce_posix_SharedCode.h
index 13724ba7c..f8892081c 100644
--- a/modules/juce_core/native/juce_posix_SharedCode.h
+++ b/modules/juce_core/native/juce_posix_SharedCode.h
@@ -989,7 +989,11 @@ bool Thread::setThreadPriority (void* handle, int priority)
     if (pthread_getschedparam ((pthread_t) handle, &policy, &param) != 0)
         return false;
 
+   #if JUCE_LINUX
+    policy = priority < 9 ? SCHED_OTHER : SCHED_RR;
+   #else
     policy = priority == 0 ? SCHED_OTHER : SCHED_RR;
+   #endif
 
     const int minPriority = sched_get_priority_min (policy);
     const int maxPriority = sched_get_priority_max (policy);
